File: app/src/main/java/com/cyl/musiclake/ui/music/list/contract/PlaylistDetailContract.java
package com.cyl.musiclake.ui.music.list.contract;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.drawable.Drawable;

import com.cyl.musiclake.base.BasePresenter;
import com.cyl.musiclake.base.BaseView;
import com.cyl.musiclake.bean.Music;
import com.cyl.musiclake.bean.Playlist;

import java.util.List;

public interface PlaylistDetailContract {

    interface View extends BaseView {

        Context getContext();

        void showPlaylistSongs(List<Music> songList);

        void showPlaylistArt(Drawable playlistArt);

        void showPlaylistArt(Bitmap bitmap);

        void removeMusic(int position);

        void success(int type);
    }

    interface Presenter extends BasePresenter<View> {

        void loadPlaylistSongs(String playlistID);

        void loadPlaylistArt(String playlistID);

        void deletePlaylist(Playlist playlist);

        void renamePlaylist(Playlist playlist, String title);

        void uncollectMusic(String pid, int position, Music music);
    }
}

File: app/src/main/java/com/cyl/musiclake/ui/music/playlist/detail/PlaylistDetailContract.kt
package com.cyl.musiclake.ui.music.playlist.detail

import com.cyl.musiclake.bean.Album
import com.cyl.musiclake.bean.Artist
import com.music.lake.musiclib.bean.BaseMusicInfo
import com.cyl.musiclake.bean.Playlist
import com.cyl.musiclake.ui.base.BaseContract

interface PlaylistDetailContract {

    interface View : BaseContract.BaseView {

        fun showPlaylistSongs(songList: MutableList<BaseMusicInfo>?)

        fun showTitle(title: String) {}

        fun showCover(cover: String) {}

        fun showDescInfo(title: String) {}

        fun removeMusic(position: Int) {}

        fun success(type: Int)

        fun showEmptyView(msg: String)
        //显示歌单异常，提示
        fun showErrorTips(msg: String, hasTry: Boolean)
    }

    interface Presenter : BaseContract.BasePresenter<View> {
        fun loadPlaylistSongs(playlist: Playlist)

        fun loadArtistSongs(artist: Artist)

        fun loadAlbumSongs(album: Album)

        fun deletePlaylist(playlist: Playlist)

        fun renamePlaylist(playlist: Playlist, title: String)
    }
}
